feat: implement A2AServer dependency checking for agents

dwmkerr opened the pull request:
## Summary
- Add A2AServer dependency checking to ensure agents wait for their owning A2AServer to be ready
- Fix A2AServer controller to not delete agents when discovery fails - follow K8s pattern where resources persist with status reflecting reality
- Add event notifications for both ready and not-ready states for better observability

## What this fixes

Currently, agents owned by A2AServers can show as Ready even when their A2AServer is not functional. This causes queries to fail when targeting these agents. With this change:
- Agents will show phase "Pending" when their owning A2AServer is not ready
- Agents automatically transition to "Ready" when A2AServer becomes ready
- Agents persist through A2AServer discovery failures (proper K8s pattern)
- Clear events are emitted for both states

dwmkerr committed:
feat: implement A2AServer dependency checking for agents

- Add checkA2AServerDependency function to validate A2AServer Ready condition
- Watch A2AServer events and trigger agent reconciliation when status changes
- Set agent phase to Pending when owning A2AServer is not Ready
- Fix A2AServer controller to not delete agents on discovery failure
- Follow Kubernetes pattern: resources persist, status reflects reality

dwmkerr committed:
feat: add A2AServerReady event when dependency is satisfied

- Emit Normal event when A2AServer becomes ready for better observability
- Complements the Warning event when A2AServer is not ready

dwmkerr committed:
fix: resolve linting issues in A2AServer dependency implementation

- Reduce cognitive complexity by extracting helper functions
- Fix gofumpt formatting issues
- Simplify nested conditionals
- Remove unused deleteAgentByA2AServer function
- Add sample YAML for A2AServer-Agent dependency

dwmkerr committed:
fix: separate registry hostname from full path for docker login

The docker/login-action requires just the registry hostname (ghcr.io)
not the full path with namespace (ghcr.io/owner/repo). This fix:
- Adds separate output for registry hostname (always ghcr.io)
- Uses hostname for docker login
- Uses full path with namespace for image tags

adriander commented on ark/internal/controller/agent_controller.go:
```diff
@@ -142,6 +151,55 @@ func (r *AgentReconciler) checkToolDependencies(ctx context.Context, agent *arkv
+
+		return r.validateA2AServerDependency(ctx, agent, ownerRef)
+	}
+
```
personal taste. I find continue harder to read and like to avoid it
eg
```
for _, ownerRef := range agent.GetOwnerReferences() {
		if ownerRef.Kind == "A2AServer" && ownerRef.APIVersion == "ark.mckinsey.com/v1prealpha1" {
			return r.validateA2AServerDependency(ctx, agent, ownerRef)
		}
	}
	```

dwmkerr committed:
fix: dynamically extract registry hostname from registry path

The registry hostname is now extracted from the full registry path
to support custom registries beyond ghcr.io. This handles cases like:
- ghcr.io/owner/repo -> ghcr.io
- docker.io/myorg/myrepo -> docker.io
- myregistry.com:5000/path -> myregistry.com:5000

dwmkerr committed:
Merge branch 'main' into feat/a2aserver-agent-dependencies
